src/AsseticBundle/Service.php

```diff
@@ -238,10 +238,15 @@ public function setupRenderer(Renderer $renderer)
     {
         $controllerConfig = $this->getControllerConfig();
         $actionConfig = $this->getActionConfig();
+        $routerConfig = $this->getRouterConfig();
+
         $config = array_merge($controllerConfig, $actionConfig);
 
-        if (count($config) == 0) {
-            $config = $this->getRouterConfig();
+        $mergeConfigs = $this->configuration->canMergeActionAndRouterConfig();
+        if ($mergeConfigs) {
+            $config = array_merge($routerConfig, $config);
```

Comment:
How would you verify that merging returns the expected result?

Reply:
I use this in my project on production and all works as expected. 
If you about tests, OK, I'll add some.